Source/WebCore/platform/gstreamer/GStreamerQuirks.h

```diff
@@ -44,7 +46,30 @@ class GStreamerQuirkBase {
     GStreamerQuirkBase() = default;
     virtual ~GStreamerQuirkBase() = default;
 
-    virtual const char* identifier() = 0;
+    virtual const char* identifier() const = 0;
+
+    // Interface of classes supplied to MediaPlayerPrivateGStreamer to store values that the quirks will need for their job.
+    class GStreamerQuirkState {
+    public:
+        GStreamerQuirkState()
+            : m_owner(nullptr)
+        {
+        }
+
+        GStreamerQuirkState(const void* owner)
+            : m_owner(owner)
+        {
+        }
+
+        virtual ~GStreamerQuirkState() = default;
+        bool isOwnedBy(const void* owner) { return m_owner == owner; }
+        bool isOwned() { return m_owner; }
+
+    private:
+        // For identification purposes only, to avoid the wrong GStreamerQuirkBase subclass to use a state that
+        // doesn't belong to itself. The pointer shouldn't be used to access the instance.
+        const void* m_owner = nullptr;
```

Comment:
If you use what I suggested above, this is no longer needed.

Reply:
Yes, but see my argument about why allowing a single state per player makes sense for this specific use case.

Reply:
In the end, I used the HashMap that you suggested and this isn't needed.